### ISSUE TITLE
Caps lock does not work

### ISSUE THREAD

**Issue by GaryMonocle** [moonlight-stream/moonlight-qt#575]
**Describe the bug**
The CAPS LOCK key ha no effect.

**Steps to reproduce**
Start a Moonlight session on a game launcher that will let you use Windows itself before starting the game itself.
I use Star Citizen RSI launcher.
Note that this is how I found the bug, but it doesn't mean it works better with games that don't use launchers.

**Screenshots**
Not applicable.

**Affected games**
I use Star Citizen RSI launcher.
The launcher starts but let's me use Windows where I can verify in notepad that caps lock doesn't work either.

**Other Moonlight clients**
- Does the issue occur when using Moonlight on iOS or Android?
-NO

**Moonlight settings (please complete the following information)**
- Have any settings been adjusted from defaults? NO
- If so, which settings have been changed? N/A
- Does the problem still occur after reverting settings back to default? N/A

**Gamepad-related issues (please complete if problem is gamepad-related)**
- Do you have any gamepads connected to your host PC directly? NO
- Does the problem still remain if you stream the desktop and use https://html5gamepad.com to test your gamepad?
  - Instructions for streaming the desktop can be found here: https://github.com/moonlight-stream/moonlight-docs/wiki/Setup-Guide

**Client PC details (please complete the following information)**
 - OS: Windows 10.0.19041 HOME
 - Moonlight Version: 3.1.1
 - GPU: On host machine: NVIDIA GEFORCE 1660 Ti

**Server PC details (please complete the following information)**
 - OS: Windows 10.0.19041 HOME
 - GeForce Experience version: 3.22.0.32
 - Nvidia GPU driver: 466.11
 - Antivirus and firewall software: Windows Defender and Windows Firewall

**Moonlight Logs (please attach)**
See attachment

**Additional context**

I noticed this issue while investigating another issue that seems related but I'm not sure how to reproduce that 2nd issue.
That 2nd was found using the Star Citizen RSI launcher too: The keyboard seemed to be stuck in SHIFT PRESSED state. All keys pressed would return their SHIFT+KEY value. For letters I would get the upper case. For keys with multiple characters, I would get the SHIFT+KEY character.
Pressing shift or caps lock had no effect.

I had a third issue during the same game session. I was then using DCS World and the keyboard just stopped working. I had to close Moonlight on the client PC to fix the issue. But it happened again later.

If I can reproduce these 2nd and 3rd issue consistently, I will log them.

**Comment by cgutman** [comment]
I can't reproduce this issue. Caps Lock works just fine for me with a GFE 3.22.0.32 host running Win10 19042 and a GFE 3.21.0.36 host running Win8.1. I tried from clients running Win10 19042 and macOS 11.2.3.

To test: I streamed my desktop, opened notepad, and tried typing with and without Caps Lock.

Does this always happen? Do you have anything on your client or host like AutoHotkey or similar programs that could be screwing with keyboard input?

Can you try rebooting your host, then [streaming the full desktop](https://github.com/moonlight-stream/moonlight-docs/wiki/Setup-Guide#using-moonlight-to-stream-your-entire-desktop) and see if it happens there?

**Comment by OskarZyg** [comment]
Same issue here.
Client (macOS) & Server (Sunshine, Windows 11) on latest releases as of 15/11/2024.
Both are set to British keyboard layouts, maybe it's worth noting the MacBook client has a physical US keyboard.
No game registers caps lock.
